make ci-lint fails on macOS

## Description

Follow-up to an issue issue I had with completing the checklist for #3994. 

## Steps to Reproduce

1. Be on macOS
2. `make ci-lint`

## Expected Behavior
The command works.

## Actual Behavior
```
devops/scripts/dev-shell-ci run make --keep-going lint typelint
devops/scripts/dev-shell-ci: line 23: sha256sum: command not found
make: *** [ci-lint] Error 127
```

## Comments

`sha256sum` is not installed by default on macOS; however, there are plenty of other built-in tools that allow one to compute hashes. We should probably create a function that chooses the appropriate hashing tool depending on the platform it's running on.

--- a/devops/scripts/dev-shell-ci
+++ b/devops/scripts/dev-shell-ci
@@ -14,20 +14,27 @@ CHECKSUM_FILE=./.linting.checksum
 
 cd "${TOPLEVEL}"
 
+# sha256sum is not included with macOS, but we can achieve the same result by
+# aliasing it to shasum, which is included by default.
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    sha256sum="shasum -a 256"
+else
+    sha256sum="sha256sum"
+fi
 
 function gen_checksums() {
     rm "$CHECKSUM_FILE" 2> /dev/null || true
 
         # shellcheck disable=SC2013
     for file in $(cat $DOCKERFILE.deps); do
-        sha256sum "$file" >> "$CHECKSUM_FILE"
+        $sha256sum "$file" >> "$CHECKSUM_FILE"
     done
 
-    sha256sum "$DOCKERFILE" >> "$CHECKSUM_FILE"
+    $sha256sum "$DOCKERFILE" >> "$CHECKSUM_FILE"
 }
 
 function check_checksum {
-    sha256sum .linting.checksum | awk '{ print $1 }'
+    $sha256sum .linting.checksum | awk '{ print $1 }'
 }
 
 function docker_run() {

